packages/bundler/src/modules/MempoolManager.ts

```diff
@@ -92,15 +92,15 @@ export class MempoolManager {
       this.mempool[index] = entry
     } else {
       debug('add userOp', userOp.sender, userOp.nonce)
-      this.incrementEntryCount(userOp.sender)
+      this.checkReputation(senderInfo, paymasterInfo, factoryInfo, aggregatorInfo)
+      this.checkMultipleRolesViolation(userOp)
       if (userOp.paymaster != null) {
         this.incrementEntryCount(userOp.paymaster)
       }
+      this.incrementEntryCount(userOp.sender)
```

Comment:
Why move the `incrementEntryCount(sender)` to be below `incrementEntryCount(paymaster)`? Seems like an unnecessary change to me.

Reply:
don't increment before you solved all cases where it can crash (otherwise, you need to catch to undo increments)